src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,623 @@
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code "SUPPRESS CHECKSTYLE (\w+)"}.
+ * </li>
+ * <li>
+ * Property {@code checkPattern} - Specify check name pattern to suppress.
+ * Property can also be a RegExp group index at {@code nearbyTextPattern} in
+ * format of {@code $x} and be picked from target file.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code ".*"}.
+ * </li>
+ * <li>
+ * Property {@code messagePattern} - Specify check violation message pattern to suppress.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code idPattern} - Specify check ID pattern to suppress.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code lineRange} - Specify negative/zero/positive value that
+ * defines the number of lines preceding/at/following the suppressing nearby text.
+ * Property can also be a RegExp group index at {@code nearbyTextPattern} in
+ * format of {@code $x} and be picked from target file.
+ * Type is {@code java.lang.String}.
+ * Default value is {@code "0"}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the filter to suppress audit events on the same line:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;Checker&quot;&gt;
+ *   &lt;module name=&quot;SuppressWithNearbyTextFilter&quot;/&gt;
+ *   &lt;module name=&quot;TreeWalker&quot;&gt;
+ *     &lt;module name=&quot;MagicNumber&quot;/&gt;
+ *   &lt;/module&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * int a = 1; // SUPPRESS CHECKSTYLE the reason for this is..
+ * int b = 1; // violation
+ * </pre>
+ * <p>
+ * To configure the filter to suppress audit events on any line that contains
+ * {@code DO NOT CHECK THIS LINE}:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;Checker&quot;&gt;
+ *   &lt;module name=&quot;SuppressWithNearbyTextFilter&quot;&gt;
+ *     &lt;property name=&quot;nearbyTextPattern&quot; value=&quot;DO NOT CHECK THIS LINE&quot;/&gt;
+ *   &lt;/module&gt;
+ *   &lt;module name=&quot;TreeWalker&quot;&gt;
+ *     &lt;module name=&quot;MagicNumber&quot;/&gt;
+ *   &lt;/module&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * int a = 1; // DO NOT CHECK THIS LINE
+ * int b = 1; // violation
+ * </pre>
+ * <p>
+ * To configure the filter to suppress audit events only on {@code FileTabCharacter}:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;Checker&quot;&gt;
+ *   &lt;module name=&quot;SuppressWithNearbyTextFilter&quot;&gt;
+ *     &lt;property name=&quot;checkPattern&quot; value=&quot;FileTabCharacter&quot;/&gt;
+ *     &lt;property name=&quot;nearbyTextPattern&quot; value=&quot;.*&quot;/&gt;
+ *   &lt;/module&gt;
+ *   &lt;module name=&quot;FileTabCharacter&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * CREATE TABLE Persons (
+ *    LastName varchar(255), -- ok
+ *    FirstName varchar(255), -- ok
```

Comment:
This example is hard to see because tab is a whitespace character. Either more explanation is needed, or another check needs to be used.